code/game/objects/items/storage/fancy.dm

```diff
@@ -236,9 +236,16 @@
 	atom_storage.set_holdable(list(/obj/item/clothing/mask/cigarette, /obj/item/lighter))
 	register_context()
 
+/obj/item/storage/fancy/cigarettes/attack(mob/living/M, mob/living/user, params)
+	. = ..()
+	if(M != user) // you can quickly put a cigarette in your mouth only
+		return ..()
+	quick_remove_item(/obj/item/clothing/mask/cigarette, user, equip = 1)
```

Comment:
You should use `interact_with_atom`, rather than `attack`. Basically the same thing but easier to work with.

Reply:
👍